pkg/operation/botanist/controlplane.go

```diff
@@ -880,9 +900,31 @@ func (b *Botanist) DeployKubeAPIServer(ctx context.Context) error {
 
 	defaultValues["admissionPlugins"] = admissionPlugins
 
+	// TODO: remove when all Gardener supported versions are >= 1.18
+	if foundDeployment && metav1.HasAnnotation(deployment.ObjectMeta, "checksum/secret-vpn-seed") {
+		vpnSecretNamesToDelete := []string{"vpn-seed", "vpn-seed-tlsauth", "vpn-shoot"}
+		for _, secret := range vpnSecretNamesToDelete {
+			err := b.K8sSeedClient.Client().Delete(context.TODO(), func(name, namespace string) *corev1.Secret {
```

Comment:
Why are we doing this here in the `DeployKubeAPIServer` function? I would have expected it in `pkg/operation/botanist/secrets.go` where the secrets are generated. Running kube-apiserver pods should not be affected and keep running.

Reply:
I though that since these are controlplane related secrets we should do it here, the kube-apiserver has a vpn sidecar after all.

Reply:
All secret computations are happening in the mentioned file - this is now mixing it up. If there is no strong reason for doing it here it would be great if can keep it altogether in `secrets.go`.

Reply:
I am also checking the annotations in the deployment here to avoid a GET call to see if it exists.